flask_jwt_extended/tokens.py

```diff
@@ -26,15 +30,19 @@ def _encode_jwt(additional_token_data, expires_delta, secret, algorithm,
     # and the 'exp' claim is not set.
     if expires_delta:
         token_data['exp'] = now + expires_delta
+
+    # Make sure additional_token_data is in conflict with default claims
+    _check_claims(['iat', 'nbf', 'jti', 'exp'], additional_token_data)
```

Comment:
I think it would be less prone to breaking with future updates if this actually compared the keys in the newly created token with the additional token data, so that these hard coded lists could go away.